pkgs/misc/themes/adapta/default.nix

```diff
@@ -1,14 +1,63 @@
-{ stdenv, fetchFromGitHub, autoreconfHook, pkgconfig, parallel, sassc, inkscape, libxml2, glib, gdk_pixbuf, librsvg, gtk-engine-murrine, gnome3 }:
+{ stdenv, fetchFromGitHub, autoreconfHook, pkgconfig, parallel, sassc, inkscape
+, libxml2, glib, gdk_pixbuf, librsvg, gtk-engine-murrine, gnome3
 
-stdenv.mkDerivation rec {
+, # Enable Gnome Shell support
+  enableGnome ? true
+
+, # Enable Cinnamon support
+  enableCinnamon ? true
+
+, # Enable Gnome Flashback support
+  enableFlashback ? true
+
+, # Enable XFCE support
+  enableXfce ? true
+
+, # Enable MATE support
+  enableMate ? true
+
+, # Enable Openbox support
+  enableOpenbox ? true
+
+, # Enable GTK+ 4.0 support
+  enableGtkNext ? false
+
+, # Enable Plank support
+  enablePlank ? false
+
+, # Enable Telegram 1.0 support
+  enableTelegram ? false
+
+, # Enable Tweetdeck-colordeck support
+  enableTweetdeck ? false
+
+, # Primary color for 'selected-items' (Default: #3F51B5 = Indigo500)
+  withSelectionColor ? null
+
+, # Secondary color for notifications and OSDs (Default: #7986CB = Indigo300)
+  withAccentColor ? null
+
+, # Secondary color for 'suggested' buttons (Default: #673AB7 = DPurple500)
+  withSuggestionColor ? null
+
+, # Tertiary color for 'destructive' buttons (Default: #F44336 = Red500)
+  withDestructionColor ? null
+}:
+
+let
+  enableFlag = flag: en: stdenv.lib.optional en "--enable-${flag}";
```

Comment:
Why not

```nix
enableFlag = flag: en: if en then "--enable-${flag}" else "--disable-"${flag}"
```
?

I'm pretty sure I've seen that pattern before, maybe its already in lib somewhere. Otherwise maybe it should be.

Reply:
Ah, yes. I'll fix.

https://github.com/NixOS/nixpkgs/blob/master/lib/strings.nix#L424